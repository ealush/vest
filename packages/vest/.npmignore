src
playground
config
testUtils
./jest.config.js
dist