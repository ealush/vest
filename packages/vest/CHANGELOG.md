# vest - Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).

## 1.0.31 - 2021-07-02
### Fixed and improved
- e6ea7d6  support global replaces (undefined)
- 34e0414  improved conditions (undefined)
- 26c28c6  all tests pass (undefined)
- 33f4e46  release (undefined)
- 6fe40c7  better bundle (undefined)
- c2cfb65  better typing (undefined)
- c6387ab  before ts settings (undefined)
- c0e9708  generate correct d.ts file (undefined)
- packages/vest/types/vest.development.d.ts
- 8e01b8e  x (undefined)
- afb3960  x (undefined)
- e0a8463  add changelog support (undefined)
- cc46c38  current (undefined)
- 4a2073f  patch(vest): changes (ealush)
- c0a1705  patch(vest): changes (ealush)
- 240386c  better create generic (ealush)
- b6db1c6  transform any to unknowns (ealush)
- f2c20a7  add done integration tests (ealush)
- ba56e33  produce methods (ealush)
- 0638243  better produce integration tests (ealush)
- 4098a11  getFailures as integration tests (ealush)
- f56cf7f  pd (ealush)
- 2ea440e  pd (ealush)
- b3032ef  relocate draftresult (ealush)
- 962141f  test getFailuresByGroup (ealush)
- e163c56  test getFailures (ealush)
- a92b6df  test collectFailureMessages (ealush)
- 81aad51  fix most tests (ealush)
- c149e90  simplify collectFailureMessages (ealush)
