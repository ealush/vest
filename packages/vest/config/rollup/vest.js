const addEsmDir = require('../../../../config/rollup/addEsmDir');
const genConfig = require('../../../../config/rollup/genConfig');
const writeMainTemplate = require('../../../../config/rollup/writeMainTemplate');
const { packageDist, packageNames } = require('../../../../util');

addEsmDir(packageDist(packageNames.VEST));
writeMainTemplate(packageDist(packageNames.VEST), packageNames.VEST);

export default genConfig({
  distPath: packageDist(packageNames.VEST),
  esInput: 'vest.mjs.js',
  input: 'vest.js',
  libraryName: packageNames.VEST,
});
