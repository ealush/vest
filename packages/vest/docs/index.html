<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <title>Vest - Validation Test</title>
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" />
    <meta name="description" content="Validation Test" />
    <meta
      name="viewport"
      content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0"
    />
    <link rel="icon" href="./_assets/favicon.ico" />
    <link
      rel="stylesheet"
      href="https://cdn.jsdelivr.net/npm/docsify-themeable@0/dist/css/theme-simple.css"
    />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/prismjs@1.22.0/themes/prism-tomorrow.css">
  </head>
  <body>
    <div id="app"></div>
    <script>
      window.$docsify = {
        name: 'vest',
        logo: '/_assets/vest-logo.png',
        repo: 'https://github.com/ealush/vest',
        search: 'auto',
        loadSidebar: true,
        subMaxLevel: 2,
        themeColor: '#5BA9D8',
      };
    </script>
    <script src="//unpkg.com/docsify/lib/docsify.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/docsify-themeable@0"></script>
    <script src="//unpkg.com/docsify/lib/plugins/search.min.js"></script>
    <script src="//unpkg.com/vest"></script>
    <script src="./_assets/index.js"></script>
  </body>
</html>
