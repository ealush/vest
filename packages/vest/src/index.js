import any from "anyone/any";
import enforce from "n4s/dist/enforce.min";
import { VERSION } from "./constants";
import test from "./core/test";
import validate from "./core/validate";
import { draft, only, skip, warn } from "./hooks";
import { singleton, runWithContext } from "./lib";

export default singleton.register({
  Enforce: enforce.Enforce,
  VERSION,
  any,
  draft,
  enforce,
  only,
  runWithContext,
  skip,
  test,
  validate,
  warn,
});
