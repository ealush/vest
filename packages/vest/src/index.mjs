import enforce from '../../n4s/src/enforce';
import create from './core/suite/create';
import test from './core/test';
import { draft, only, skip, warn, group } from './hooks';

const VERSION = VEST_VERSION;

export default {
  VERSION,
  create,
  draft,
  enforce,
  group,
  only,
  skip,
  test,
  warn,
};

export { VERSION, create, draft, enforce, group, only, skip, test, warn };
