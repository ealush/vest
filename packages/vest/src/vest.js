import create from 'createSuite';
import enforce from 'enforce';
import { only, skip, warn, group } from 'hooks';
import test from 'test';

const VERSION = __LIB_VERSION__;

export default {
  VERSION,
  create,
  enforce,
  group,
  only,
  skip,
  test,
  warn,
};
