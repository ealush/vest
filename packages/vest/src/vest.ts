import { enforce } from 'n4s';

import create from 'create';
import context from 'ctx';
import each from 'each';
import { only, skip } from 'exclusive';
import group from 'group';
import include from 'include';
import { eager } from 'mode';
import omitWhen from 'omitWhen';
import optional from 'optionalTests';
import skipWhen from 'skipWhen';
import { test } from 'test';
import warn from 'warn';

const VERSION = __LIB_VERSION__;
export {
  test,
  create,
  each,
  only,
  skip,
  warn,
  group,
  optional,
  skipWhen,
  omitWhen,
  enforce,
  VERSION,
  context,
  include,
  eager,
};
