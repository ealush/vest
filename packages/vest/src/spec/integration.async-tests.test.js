import vest from '..';

function genValidate({ create, test, enforce, ...vest }) {
  return create('suite_name', () => {
    test('field_1', 'field_statement_1', () => false);
    test('field_2', 'field_statement_2', () => {
      enforce(2).equals(3);
    });
    test('field_3', 'field_statement_3', () => {});
    test('field_4', 'field_statement_4', () => {
      vest.warn();
      throw new Error();
    });
    test('field_4', 'field_statement_4', () => {
      vest.warn();
    });
    test('field_5', 'field_statement_5', () => false);
    test('field_5', 'field_statement_6', () => false);
    test(
      'field_6',
      'async_statement_1',
      () =>
        new Promise(res => {
          setTimeout(res, 250);
        })
    );
    test('field_7', () => Promise.reject('async_statement_2'));
  });
}

let validate;
describe('Stateful behavior', () => {
  let result, callback_1, callback_2, callback_3, control;

  beforeEach(() => {
    validate = genValidate(vest);
  });

  beforeAll(() => {
    callback_1 = jest.fn();
    callback_2 = jest.fn();
    callback_3 = jest.fn();
    control = jest.fn();
  });

  test.skipOnWatch(
    'Should have all fields',
    () =>
      new Promise(done => {
        // ❗️Why is this test async? Because of the `resetState` beforeEach.
        // We must not clean up before the suite is actually done.
        result = validate(vest).done(done);
        expect(result.tests).toHaveProperty('field_1');
        expect(result.tests).toHaveProperty('field_2');
        expect(result.tests).toHaveProperty('field_4');
        expect(result.tests).toHaveProperty('field_5');
        expect(result.tests).toHaveProperty('field_6');
        expect(result.tests).toHaveProperty('field_7');
        expect(result.hasErrors('field_7')).toBe(false);
        expect(result.tests).toMatchSnapshot();
      })
  );

  it('Should invoke done callback specified with sync field immediately, and the others after finishing', () =>
    new Promise(done => {
      result = validate(vest);
      result
        .done('field_1', callback_1)
        .done('field_6', callback_2)
        .done(callback_3);
      expect(callback_1).toHaveBeenCalled();
      expect(callback_2).not.toHaveBeenCalled();
      expect(callback_3).not.toHaveBeenCalled();

      setTimeout(() => {
        expect(callback_2).not.toHaveBeenCalled();
        expect(callback_3).not.toHaveBeenCalled();
        expect(result.hasErrors('field_7')).toBe(true);
        control();
      });

      setTimeout(() => {
        expect(callback_2).toHaveBeenCalled();
        expect(callback_3).toHaveBeenCalled();
        expect(control).toHaveBeenCalled();
        done();
      }, 250);
    }));
});
