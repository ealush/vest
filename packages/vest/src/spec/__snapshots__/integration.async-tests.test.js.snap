// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stateful behavior Should have all fields 1`] = `
Object {
  "field_1": Object {
    "errorCount": 1,
    "errors": Array [
      "field_statement_1",
    ],
    "testCount": 1,
    "warnCount": 0,
  },
  "field_2": Object {
    "errorCount": 1,
    "errors": Array [
      "field_statement_2",
    ],
    "testCount": 1,
    "warnCount": 0,
  },
  "field_3": Object {
    "errorCount": 0,
    "testCount": 1,
    "warnCount": 0,
  },
  "field_4": Object {
    "errorCount": 0,
    "testCount": 2,
    "warnCount": 1,
    "warnings": Array [
      "field_statement_4",
    ],
  },
  "field_5": Object {
    "errorCount": 2,
    "errors": Array [
      "field_statement_5",
      "field_statement_6",
    ],
    "testCount": 2,
    "warnCount": 0,
  },
  "field_6": Object {
    "errorCount": 0,
    "testCount": 1,
    "warnCount": 0,
  },
  "field_7": Object {
    "errorCount": 0,
    "testCount": 1,
    "warnCount": 0,
  },
}
`;

exports[`Stateful behavior Should have all fields 2`] = `
Object {
  "field_1": Object {
    "errorCount": 1,
    "errors": Array [
      "field_statement_1",
    ],
    "testCount": 1,
    "warnCount": 0,
  },
  "field_2": Object {
    "errorCount": 1,
    "errors": Array [
      "field_statement_2",
    ],
    "testCount": 1,
    "warnCount": 0,
  },
  "field_3": Object {
    "errorCount": 0,
    "testCount": 1,
    "warnCount": 0,
  },
  "field_4": Object {
    "errorCount": 0,
    "testCount": 2,
    "warnCount": 1,
    "warnings": Array [
      "field_statement_4",
    ],
  },
  "field_5": Object {
    "errorCount": 2,
    "errors": Array [
      "field_statement_5",
      "field_statement_6",
    ],
    "testCount": 2,
    "warnCount": 0,
  },
  "field_6": Object {
    "errorCount": 0,
    "testCount": 1,
    "warnCount": 0,
  },
  "field_7": Object {
    "errorCount": 0,
    "testCount": 1,
    "warnCount": 0,
  },
}
`;

exports[`Stateful behavior Should have all fields 3`] = `
Object {
  "field_1": Object {
    "errorCount": 1,
    "errors": Array [
      "field_statement_1",
    ],
    "testCount": 1,
    "warnCount": 0,
  },
  "field_2": Object {
    "errorCount": 1,
    "errors": Array [
      "field_statement_2",
    ],
    "testCount": 1,
    "warnCount": 0,
  },
  "field_3": Object {
    "errorCount": 0,
    "testCount": 1,
    "warnCount": 0,
  },
  "field_4": Object {
    "errorCount": 0,
    "testCount": 2,
    "warnCount": 1,
    "warnings": Array [
      "field_statement_4",
    ],
  },
  "field_5": Object {
    "errorCount": 2,
    "errors": Array [
      "field_statement_5",
      "field_statement_6",
    ],
    "testCount": 2,
    "warnCount": 0,
  },
  "field_6": Object {
    "errorCount": 0,
    "testCount": 1,
    "warnCount": 0,
  },
  "field_7": Object {
    "errorCount": 0,
    "testCount": 1,
    "warnCount": 0,
  },
}
`;

exports[`Stateful behavior Should have all fields 4`] = `
Object {
  "field_1": Object {
    "errorCount": 1,
    "errors": Array [
      "field_statement_1",
    ],
    "testCount": 1,
    "warnCount": 0,
  },
  "field_2": Object {
    "errorCount": 1,
    "errors": Array [
      "field_statement_2",
    ],
    "testCount": 1,
    "warnCount": 0,
  },
  "field_3": Object {
    "errorCount": 0,
    "testCount": 1,
    "warnCount": 0,
  },
  "field_4": Object {
    "errorCount": 0,
    "testCount": 2,
    "warnCount": 1,
    "warnings": Array [
      "field_statement_4",
    ],
  },
  "field_5": Object {
    "errorCount": 2,
    "errors": Array [
      "field_statement_5",
      "field_statement_6",
    ],
    "testCount": 2,
    "warnCount": 0,
  },
  "field_6": Object {
    "errorCount": 0,
    "testCount": 1,
    "warnCount": 0,
  },
  "field_7": Object {
    "errorCount": 0,
    "testCount": 1,
    "warnCount": 0,
  },
}
`;
